[WTH-44] 게시판 테스트 코드 작성

jj0526 opened the pull request:
## PR 내용
- Notice 테스트 코드 작성
  - 저장(service)
  - 조회(repository)
  - 검색(repository)
<br>

## PR 세부사항
- Notice의 `NoticeSaveService`의 `save` 기능, `NoticeRepository`의 `findPageBy` (조회) 기능,  `NoticeRepository`의 `search` 기능의 테스트 코드를 작성하였습니다

## 질문
- 한 메소드의 여러 케이스들과 엣지 케이스들을 모두 추가하는게 좋을까요?
<br>

## 관련 스크린샷
- x
<br>

## 주의사항
- x
<br>

## 체크 리스트

- [x] 리뷰어 설정
- [x] Assignee 설정
- [x] Label 설정
- [x] 제목 양식 맞췄나요? (ex. #0 Feat: 기능 추가)
- [x] 변경 사항에 대한 테스트

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Tests**
  * Notice 도메인의 저장소, 서비스, 유스케이스에 대한 통합 및 단위 테스트 추가
  * 공지사항 페이징 조회 및 검색 결과의 정렬·페이징 동작 검증 테스트 추가
  * 저장 동작 검증 및 연관 사용자 매핑 검증 테스트 추가
  * 다양한 파일 존재 여부 케이스를 반영한 유스케이스 단위 테스트 추가
  * 테스트 데이터 생성 자동화를 위한 Notice 픽스처 추가
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

hyxklee committed:
config: TestContainer 및 test 환경 설정

jj0526 committed:
test: notice 저장 테스트 코드 작성

jj0526 committed:
test: notice 조회 테스트 코드 작성

jj0526 committed:
test: notice 검색 테스트 코드 작성

jj0526 committed:
style: 개행 추가

jj0526 committed:
test: NoticeSaveService 통합 테스트로 변경 (@DataJpaTest + Service 포함)

jj0526 committed:
test: Notice 테스트 코드 작성을 위한 NoticeFixture 추가

jj0526 committed:
test: Notice 공지사항이 최신순 정렬 확인 테스트 코드 작성

jj0526 committed:
test: 공지사항 검색시 결과와 파일 존재여부가 정상적으로 반환하는지 확인하는 테스트 코드 작성

jj0526 committed:
test: 공지사항 최신순 정렬 테스트 코드 수정
- fileGetService mock 설정을 nullable(Long) → any()로 변경
- hasNext() 정상 반환 검증 추가

seokjun01 commented on src/test/java/leets/weeth/domain/board/application/usecase/NoticeUsecaseImplTest.java:
```diff
@@ -0,0 +1,178 @@
+        when(noticeFindService.findRecentNotices(any(Pageable.class))).thenReturn(slice);
+        when(fileGetService.findAllByNotice(any())).thenReturn(List.of());
+
+        when(noticeMapper.toAll(any(Notice.class), anyBoolean()))
```
given 단계에는  객체 설정 및 가짜 응답을 정의하는 단계라고 알고 있습니다! 
when-thenReturn 구문이 잘못된 것은 아닌 것 같지만, 가독성 측면에서 given-willReturn 구문으로 감싸주시는 건 어떨까요?! 😄

seokjun01 reviewed (approved):
고생하셨습니다 ~~ 궁금한 점 및 제 의견 남겨두었습니다! 👍🏻👍🏻

hyxklee reviewed (commented):
고생하셨습니당!
전반적으로 좋은 방향으로 작성되고 있으나, 조금 더 다양한 assertThat 구문을 활용해볼 수 있을 것 같아요!
또 fixture를 적극적으로 활용해 테스트 코드의 가독성을 높일 방법이 아직 많아 보여서 조금씩 더 리팩토링을 해보면 좋을 것 같습니당

jj0526 committed:
style: 개행 추가

jj0526 committed:
refactor: given-willReturn

jj0526 committed:
style: 추후 테스트 추가할 기능에 @Disabled 추가

jj0526 committed:
test: 공지사항 전체 순서 검증 추가

jj0526 commented on src/test/java/leets/weeth/domain/board/application/usecase/NoticeUsecaseImplTest.java:
```diff
@@ -0,0 +1,178 @@
+    @Mock
+    private NoticeFindService noticeFindService;
+    @Mock
+    private FileGetService fileGetService;
```
넵!

jj0526 committed:
Merge develop

jj0526 committed:
test: NoticeRepository 테스트에서 UserRepository 의존성 제거

jj0526 committed:
test: NoticeRepository 테스트에서 UserSaveService 제거

jj0526 committed:
test: NoticeFixture에 user 없이 생성 가능한 createNotice 메서드 추가 및 디렉토리 변경

jj0526 committed:
refactor: 공지 레포지터리 테스트 수정
- NoticeFixture 사용하는 반복문으로 수정
- 메서드명 테스트 목적이 포함되도록 수정
- 검증 로직 수정(id 내림차순 조회 및 검색 결과 검증)

jj0526 committed:
style: 테스트코드에서 불필요한 공백 제거

seokjun01 commented on src/test/java/leets/weeth/domain/board/domain/test/fixture/NoticeFixture.java:
```diff
@@ -0,0 +1,15 @@
+                .user(user)
+                .commentCount(0)
+                .build();
+    }
```
답변 감사합니다!

jj0526 committed:
test: NoticeSaveServiceTest 제거
